app/lib/actions/katello/orphan_cleanup/remove_orphaned_content_units.rb

```diff
@@ -0,0 +1,32 @@
+module Actions
+  module Katello
+    module OrphanCleanup
+      class RemoveOrphanedContentUnits < Actions::Base
+        def plan(options = {})
+          plan_self(id: options[:repo_id], destroy_all: options[:destroy_all])
+        end
+
+        def run
+          content_types_to_index = []
+          if input[:destroy_all]
+            ::Katello::RepositoryTypeManager.enabled_repository_types.each_value do |repo_type|
+              content_types_to_index << repo_type.content_types_to_index
+            end
+          elsif input[:id]
+            repo = ::Katello::Repository.find(input[:id])
+            content_types_to_index = repo.repository_type.content_types_to_index
+          else
+            fail "Pass either a repository to determine content type or destroy_all to destroy all orphaned content units"
```

Comment:
What do you think about removing the `destroy_all` param?  That way only `repo_id` would be needed and the logic would be a bit simplified. Orphan cleanup should only be deleting useless data, so I don't think a safety net is needed.